### PR TITLE
Formatting and cleanup.

### PR DESCRIPTION
**Pull request by olliestanley**
Signed-off-by: Ollie dzineit@live.co.uk

And by cleanup I mean removed unused imports.

### CODE REVIEW

File: src/main/java/org/spout/vanilla/event/entity/EntityDamageByBlockEvent.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.spout.vanilla.event.entity;
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/entity/EntityDamageByBlockEvent.java]: We don't want events like these, instead we use EntityDamageEvent and then see the DamageCause enum in org.spout.vanilla.controller.source.DamageCause

File: src/main/java/org/spout/vanilla/event/entity/EntityDamageEvent.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
+package org.spout.vanilla.event.entity;
+
+import org.spout.api.entity.Entity;
+import org.spout.api.event.Cancellable;
+import org.spout.api.event.HandlerList;
+import org.spout.api.event.entity.EntityEvent;
+
+public class EntityDamageEvent extends EntityEvent implements Cancellable {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/entity/EntityDamageEvent.java]: EntityDamageEvent is an event in SpoutAPI, not Vanilla

File: src/main/java/org/spout/vanilla/event/entity/EntityDamageEvent.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	public void setCancelled(boolean cancelled) {
+		this.cancelled = cancelled;
+	}
+
+	@Override
+    public HandlerList getHandlers() {
+	    return handlers;
+    }
+	
+	public enum DamageCause {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/entity/EntityDamageEvent.java]: DamageCause is already an existing enum

**Comment by olliestanley** [comment]
I looked for EntityDamageEvent and couldn't find it :0. Maybe I was doing something stupid :/. I'll fix that then leave the formatting / cleanup changes in tomorrow. Sorry.

**Comment by olliestanley** [comment]
Although the weakness of no subs for ByEntity / ByBlock is that you can't get the block / entity if it is in that case. Or do you have a different method?

**Comment by Zidane** [comment]
@PandazNWafflez

I think @W1ndwaker meant this should be SpoutAPI. To completely cover all three events in one, it would simply be...

public EntityDamageEvent(Entity e, Source damager, Source reason). Block and Entity can be sources :). The DamageCause enum in Vanilla is also a source and would work for the reason.

**Comment by olliestanley** [comment]
So, I make a PR for SpoutAPI with the EntityDamageEvent and delete the 3 files here?

**Comment by olliestanley** [comment]
And also, if the DamageCause enum is in Vanilla, how can you use it in SpoutAPI for the EntityDamageEvent?

**Comment by Zidane** [comment]
@PandazNWafflez

To your first comment, yes

To your second comment, DamageCause extends Source. Putting the reason as a Source command works :).

**Comment by olliestanley** [comment]
Ok, well I've removed the files here, I'm off to SpoutAPI :P

EDIT: BTW, didn't you used to be in the VanillaDev organisation @NinjaZidane?

**Comment by Zidane** [comment]
@PandazNWafflez

I still am? I am apart of SpoutDev and I lead VanillaDev.

**Comment by windy1** [comment]
Erm SpoutAPI already has an EntityChangeHealthEvent

**Comment by Zidane** [comment]
@W1ndwaker
hmm that is correct...I forgot how general I made these events!

**Comment by olliestanley** [comment]
Oh, and I could've sworn I looked at the members on the GitHub organisation last night and you weren't listed :/

And @W1ndwaker Well it's a good thing I never made the PR then.

But anyway, is this good to be pulled now I got rid of that stuff?

EDIT: It also doesn't list VanillaDev in your organisations on your GitHub profile :O

**Comment by windy1** [comment]
Look's good, you are going to need to rebase before this can be pulled though. http://forums.spout.org/wiki/pull-request-guidelines/#rebasing

**Comment by olliestanley** [comment]
The bane of my existence... rebasing... I use Tortoise Git.

Tortoise Git's rebasing is terrible.

**Comment by kitskub** [comment]
Try Git Extensions. I use it, and it's great.

**Comment by windy1** [comment]
Closing this due to lack of response.

**Comment by olliestanley** [comment]
Sorry, I am actually still trying to work out how to rebase. Tortoise Git seems to only let you rebase from your own repositories on GitHub, so I can't rebase to your master :/

**Comment by ZNickq** [comment]
git pull origin master --rebase in git bash ;)

**Comment by windy1** [comment]
No, that will pull from your fork @Znickq
He's working on master so that would do nothing in this case.

You want to do
`git remote add upstream git://github.com/VanillaDev/Vanilla.git && git pull --rebase upstream master && git push origin master`
You might need to force push.

If he had worked on another branch and kept his master updated and untouched he could have just pulled from that.

**Comment by Afforess** [comment]
> Sorry, I am actually still trying to work out how to rebase. Tortoise Git seems to only let you rebase from your own repositories on GitHub, so I can't rebase to your master :/\

I use TortoiseGit, and I use remote rebases all the time for some other projects. I usually rebase off the url of the project, which should update your repo's head.

**Comment by olliestanley** [comment]
@Afforess I tried adding a new URL, but it didn't show up. This is what I did:

Right Click Vanilla Folder --> TortoiseGit --> *Rebase --> ... --> Manage Remotes

I then added a new remote called master with the URL as https://github.com/VanillaDev/Vanilla, but it didn't show up in the list after. This is what I don't get.

**Comment by Afforess** [comment]
TortoiseGit -> Fetch -> Set arbitrary url -> Check "Launch Rebase After Fetch" -> Ok  -> Rebase appears

Set the branch to whatever branch you were working on (master, if none), and upstream to FETCH_HEAD. Then rebase.

**Comment by olliestanley** [comment]
@Afforess Where's Run? I got to there and I can't see a Run button :O

**Comment by Afforess** [comment]
It's actually called okay. I was going from memory. I've updated my comment.

**Comment by olliestanley** [comment]
Done.

**Comment by olliestanley** [comment]
Hmm... It was successful, but there were no changes to the PR / my repo, is that good or bad?

**Comment by windy1** [comment]
Did you push your changes @DziNeIT?

**Pull request closed without merging**